### ISSUE TITLE
Weird Behavior of Ordered Boosting on Small Dataset

### ISSUE THREAD

**Issue by zehaoli2002** [catboost/catboost#2444]
catboost version: 1.2

Hi, I am currently employing CatBoost for binary classification on an imbalanced (5:1) dataset with ~600 rows and ~30 numeric columns. I am using a modified version of log loss to deal with the imbalance. To evaluate the model, I use stratified cross validation on the label with `k` folds.

Here's the weird behavior. When `k >= 6`, I observe very good result with ordered boosting; however, when `k < 6`, the model is not learning at all (train and validation loss stuck at `0.69`). Regardless of `k`, the model learns with plain boosting. 

I get this result with everything default. I messed with the bootstrap type a bit: MVS with 0.8, Bernoulli with 0.5 ~ 0.9, and no boostrap. The result is the same.

In Figure 2 of the paper, the minimum `fraction` is different across datasets. Were similar behavior observed in your experiments?
